libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -460,7 +464,7 @@ else if (record.getBoolean(LOCALLY_UPDATED))
          * circumstances, we will do nothing and return here.
          */
         if (mergeMode == MergeMode.LEAVE_IF_CHANGED &&
-        		(action == Action.update || action == Action.delete) &&
+        		!locallyCreated &&
```

Comment:
if it is marked as created and deleted then action will be deleted, but we really shouldn't go to the server.
Ideally the app should just delete the record if it was created locally instead of waiting until sync up to do it.